pravega/pravega-samples
# [issue-162] improve the performance of Pravega based Terasort sample code

**Change log description**
Made segments of stream configurable as the existing 3 segments are too limited for large scale performance test;
Changed back to the approach using multiple streams with fixed(1) segment for terasort output

**Purpose of the change**
Fixes #162.

**What the code does**
Add optional command-line 5th parameter for TeraGen job to indicate the number of segments for the stream to be created.
`PravegaSequenceOutputFormat` is added as the subclass of `PravegaOutputFormat`, which returns `RecordWriter` with a separate stream created for it. All the event from the task will go to this particular stream only.

**How to verify it**
With Teragen, add Segments of Stream command-line parameter to change the number of segments.
With Terasort, the amount of generated streams should equal to the number of reducer tasks.

Signed-off-by: kevinhan88 <kevinhan88@gmail.com>
